hack/util.sh

```diff
@@ -641,7 +641,11 @@ function install_registry {
 }
 
 function wait_for_registry {
-	wait_for_command '[[ "$(oc get endpoints docker-registry --output-version=v1 --template="{{ if .subsets }}{{ len .subsets }}{{ else }}0{{ end }}" --config=${ADMIN_KUBECONFIG} || echo "0")" != "0" ]]' $((5*TIME_MIN))
+	local config="--config=${ADMIN_KUBECONFIG}"
+	local tmplsubsets='{{ if .subsets }}{{ len .subsets }}{{ else }}0{{ end }}'
+	wait_for_command '[[ "$(oc get endpoints docker-registry --output-version=v1 --template='"'$tmplsubsets' $config"' || echo 0)" != "0" ]]' $((5*TIME_MIN))
```

Comment:
The `wait_for_command` cannot be easily replaced in `./hack/util.sh`. The file is sourced by `./hack/cmd_util.sh` which defines `os::cmd::wait_until_text`. Even if I extract the function from the `cmd_util.sh` and source it with `./hack/util.sh`, the invocation fails due to test suit uninitialized:

    [INFO] Installing the registry
    --> Creating registry registry ...
        serviceaccount "registry" created
        clusterrolebinding "registry-registry-role" created
        deploymentconfig "docker-registry" created
        service "docker-registry" created
    --> Success
    [ERROR] jUnit test marker could not be placed, expected suites in flight, got 0
    !!! Error in /data/src/github.com/openshift/origin/hack/cmd_util.sh:69
            'return 1' exited with status 1
    Call stack:
            1: /data/src/github.com/openshift/origin/hack/cmd_util.sh:69 os::cmd::internal::run_until_text(...)
            2: /data/src/github.com/openshift/origin/hack/cmd_util.sh:133 os::cmd::try_until_text(...)
            3: /data/src/github.com/openshift/origin/hack/util.sh:343 wait_for_registry(...)
            4: ./test/extended/../../test/extended/setup.sh:115 os::test::extended::setup(...)
            5: ./test/extended/core.sh:14 main(...)
    Exiting with status 1

I agree that ideally the `wait_for_command` should be removed and replaced with the `os::cmd::*` commands but the suit seems not ready for that ATM. And my bash-fu doesn't seem to be at the level capable of doing some bigger refactor at this point.